### PR TITLE
Jwt auth

### PR DESCRIPTION
**Pull request by mjburghoffer**

### CODE REVIEW

File: soa-auth/src/main/java/com/relateiq/soa/auth/JwtProperties.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.relateiq.soa.auth;
+
+@SuppressWarnings("WeakerAccess")
+public final class JwtProperties {
+  private JwtProperties() {}
+
+  public static final String JWT_ISSUER = "auth.jwt.issuer";
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/JwtProperties.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

File: soa-auth/src/main/java/com/relateiq/soa/auth/JwtProperties.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  public static final String JWT_AUTH_TOKEN_CREDENTIAL = "auth.jwt.token";
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/JwtProperties.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

File: soa-auth/src/main/java/com/relateiq/soa/auth/JwtConstants.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.relateiq.soa.auth;
+
+@SuppressWarnings("WeakerAccess")
+public final class JwtConstants {
+  private JwtConstants() {}
+
+  public static final String ALMOST_SECRET = "something super secret that goes on for kind of a long time, why? you ask?  The Internet.";
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/JwtConstants.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  @Override
+  @Nonnull
+  public Iterator<Map.Entry<String, Object>> iterator() {
+    return asUnmodifiableMap().entrySet().iterator();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: MINOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+  @Override
+  @Nonnull
+  public Iterator<Map.Entry<String, Object>> iterator() {
+    return asUnmodifiableMap().entrySet().iterator();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: MINOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 8 issues:
* CRITICAL 2 critical
* MAJOR 1 major
* MINOR 5 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* CRITICAL [Either log or rethrow this exception.](https://github.com/relateiq/soa/blob/0f95a515da8b4236c9b7d1965c7627aa7265f3d2/soa-commons/src/main/java/com/relateiq/soa/commons/TransactionIdProvider.java#L249) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Remove this unused "FULL_TX_MASK" private field.](https://github.com/relateiq/soa/blob/0f95a515da8b4236c9b7d1965c7627aa7265f3d2/soa-commons/src/main/java/com/relateiq/soa/commons/TransactionIdProvider.java#L61) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* CRITICAL [Either log or rethrow this exception.](https://github.com/relateiq/soa/blob/0f95a515da8b4236c9b7d1965c7627aa7265f3d2/soa-commons/src/main/java/com/relateiq/soa/commons/TransactionIdProvider.java#L240) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 5 issues:
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java (hunk 4 above)
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: MINOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java (hunk 5 above)
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: MINOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: pom.xml [hunk 6]
```diff
@@ -183,9 +235,71 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (jontg) [inline pom.xml]: 👍

File: pom.xml [hunk 7]
```diff
@@ -183,9 +235,71 @@
...
+                    <exclude>org.projectlombok</exclude>
+                    <exclude>org.apache.commons:commons-lang3</exclude>
+                  </excludes>
+                </bannedDependencies>
+                <reactorModuleConvergence>
+                  <message>This reactor is not valid</message>
+                  <ignoreModuleDependencies>true</ignoreModuleDependencies>
+                </reactorModuleConvergence>
+                <requireReleaseDeps>
+                  <message>No SNAPSHOT dependencies allowed!</message>
```
Comment (jontg) [inline pom.xml]: <3

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.function.BiConsumer;
+
+/**
+ * A way to transport all non-registered JWT claims to a builder.
+ */
+public final class CustomJwtClaims implements Iterable<Map.Entry<String, Object>> {
+
+  @Nonnull private final Map<String, Object> claims = new HashMap<>();
+
+  public CustomJwtClaims() {
```
Comment (relateiq-ec2) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by jontg** [comment]
LGTM!

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * A way to transport all non-registered JWT claims to a builder.
+ */
+public final class CustomJwtClaims implements Iterable<Map.Entry<String, Object>> {
+
+  @Nonnull private final Map<String, Object> claims = new HashMap<>();
```
Comment (christydennison) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: Since you're using Guava, any reason not to use Maps.newHashMap() here?

**Comment by christydennison** [comment]
This looks so good!! :D

File: soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java (hunk 9 above)
Comment (mjburghoffer) [inline soa-auth/src/main/java/com/relateiq/soa/auth/CustomJwtClaims.java]: It takes more characters of code for literally no gain.  Truth be told I'd rather even not use Guava anywhere.

**Pull request merged**